drivers/npmx/npmx_shell.c

```diff
@@ -2290,690 +2299,481 @@ static int cmd_gpio_status_get(const struct shell *shell, size_t argc, char **ar
+static int cmd_ldsw_active_discharge_set(const struct shell *shell, size_t argc, char **argv)
 {
-	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
-
-	if (npmx_instance == NULL) {
-		shell_error(shell, "Error: shell is not initialized.");
-		return 0;
-	}
-	if (argc < 2) {
-		shell_error(shell, "Error: missing LDSW instance index and new status value.");
+	args_info_t args_info = { .expected_args = 2,
+				  .arg = {
+					  [0] = { SHELL_ARG_TYPE_UINT32_INDEX, "LDSW" },
+					  [1] = { SHELL_ARG_TYPE_BOOL_VALUE, "status" },
+				  } };
+	if (!arguments_check(shell, argc, argv, &args_info)) {
 		return 0;
 	}
-	if (argc < 3) {
-		shell_error(shell, "Error: missing new status value.");
+
+	npmx_ldsw_t *ldsw_instance = ldsw_instance_get(shell, args_info.arg[0].result.uvalue);
+	if (ldsw_instance == NULL) {
 		return 0;
 	}
 
-	int err = 0;
-	uint8_t ldsw_indx = CLAMP(shell_strtoul(argv[1], 0, &err), 0, UINT8_MAX);
-	uint8_t ldsw_set = CLAMP(shell_strtoul(argv[2], 0, &err), 0, UINT8_MAX);
+	bool discharge_status = args_info.arg[1].result.bvalue;
+	npmx_error_t err_code =
+		npmx_ldsw_active_discharge_enable_set(ldsw_instance, discharge_status);
 
-	if (err != 0) {
-		shell_error(shell, "Error: instance index and status have to be integers.");
+	if (!check_error_code(shell, err_code)) {
+		print_set_error(shell, "LDSW active discharge status");
 		return 0;
 	}
 
-	if (ldsw_indx >= NPM_LDSW_COUNT) {
-		shell_error(shell, "Error: LDSW instance index is too high: no such instance.");
+	print_success(shell, discharge_status, UNIT_TYPE_NONE);
+	return 0;
+}
+
+static int cmd_ldsw_active_discharge_get(const struct shell *shell, size_t argc, char **argv)
+{
+	args_info_t args_info = { .expected_args = 1,
+				  .arg = {
+					  [0] = { SHELL_ARG_TYPE_UINT32_INDEX, "LDSW" },
+				  } };
+	if (!arguments_check(shell, argc, argv, &args_info)) {
 		return 0;
 	}
 
-	if (ldsw_set > 1) {
-		shell_error(shell, "Error: wrong new enable value.");
+	npmx_ldsw_t *ldsw_instance = ldsw_instance_get(shell, args_info.arg[0].result.uvalue);
+	if (ldsw_instance == NULL) {
 		return 0;
 	}
 
-	npmx_ldsw_task_t ldsw_task = ldsw_set == 1 ? NPMX_LDSW_TASK_ENABLE : NPMX_LDSW_TASK_DISABLE;
-	npmx_ldsw_t *ldsw_instance = npmx_ldsw_get(npmx_instance, ldsw_indx);
-
-	npmx_error_t err_code = npmx_ldsw_task_trigger(ldsw_instance, ldsw_task);
-
-	if (check_error_code(shell, err_code)) {
-		shell_print(shell, "Success: %d.", ldsw_set);
-	} else {
-		shell_error(shell, "Error: unable to set LDSW status.");
+	bool discharge_status;
+	npmx_error_t err_code =
+		npmx_ldsw_active_discharge_enable_get(ldsw_instance, &discharge_status);
+	if (!check_error_code(shell, err_code)) {
+		print_get_error(shell, "LDSW active discharge status");
+		return 0;
 	}
 
+	print_value(shell, discharge_status, UNIT_TYPE_NONE);
 	return 0;
 }
 
-static int cmd_ldsw_get(const struct shell *shell, size_t argc, char **argv)
+static npmx_ldsw_gpio_t ldsw_gpio_index_convert(int32_t gpio_idx)
```

Comment:
Can that be merged with conversion for buck? Maybe one common function would be enough?

Reply:
It is a good idea, but not in this PR.

Reply:
NRFX-4803